specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2023-09-01/virtualMachineInstances.json

```diff
@@ -0,0 +1,1564 @@
+    },
+    "VirtualMachineInstanceStatus": {
+      "type": "object",
+      "x-ms-client-flatten": true,
+      "readOnly": true,
+      "properties": {
+        "errorCode": {
+          "type": "string",
+          "description": "VirtualMachine provisioning error code"
+        },
+        "errorMessage": {
+          "type": "string",
+          "description": "Descriptive error message"
+        },
+        "powerState": {
+          "type": "string",
+          "description": "The power state of the virtual machine instance",
+          "enum": [
+            "Deallocated",
+            "Deallocating",
+            "Running",
+            "Starting",
+            "Stopped",
+            "Stopping",
+            "Unknown"
+          ],
+          "x-ms-enum": {
+            "name": "PowerStateEnum",
+            "modelAsString": true
+          }
+        },
+        "provisioningStatus": {
+          "type": "object",
+          "properties": {
+            "operationId": {
+              "type": "string",
+              "description": "The ID of the operation performed on the virtual machine instance"
+            },
+            "status": {
+              "type": "string",
+              "description": "The status of the operation performed on the virtual machine instance [Succeeded, Failed, InProgress]",
+              "enum": [
+                "Succeeded",
+                "Failed",
+                "InProgress"
+              ],
+              "x-ms-enum": {
+                "name": "Status",
+                "modelAsString": true
+              }
+            }
+          }
+        }
+      },
+      "description": "The observed state of virtual machine instances"
+    },
+    "VirtualMachineInstance": {
+      "type": "object",
+      "x-ms-client-flatten": true,
+      "x-ms-azure-resource": true,
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/VirtualMachineInstanceProperties"
+        },
+        "extendedLocation": {
+          "$ref": "./common.json#/definitions/ExtendedLocation",
+          "description": "The extendedLocation of the resource."
+        },
+        "identity": {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/Identity"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "description": "The virtual machine instance resource definition."
+    },
+    "VirtualMachineInstanceView": {
+      "type": "object",
+      "properties": {
+        "vmAgent": {
+          "$ref": "#/definitions/VirtualMachineVMConfigAgentInstanceView",
+          "description": "The VM Config Agent running on the virtual machine."
+        }
+      },
+      "description": "The instance view of a virtual machine."
+    },
+    "VirtualMachineVMConfigAgentInstanceView": {
```

Comment:
>VirtualMachineVMConfigAgentInstanceView [](http://example.com/codeflow?start=5&length=39)

This type name seems to have the similar double VM redundancy thing going on - want to fix it?

Reply:
Confirmed that it indeed has a redundancy. Will fix.